timemachine/fe/model_rabfe.py

```diff
@@ -17,17 +18,17 @@
 class AbsoluteModel(ABC):
     def __init__(
         self,
-        client: AbstractClient or None,
+        client: Optional[AbstractClient],
         ff: Forcefield,
         host_system: openmm.System,
-        host_schedule: np.ndarray,
+        host_schedule: List[float],
```

Comment:
Mild surprise: I thought this was often an array

Reply:
Good call; I don't think we have any cases of passing an array in the timemachine code, but there are plenty of examples in the tests (we don't currently run mypy on tests; this seemed like maybe too much overhead).

I think a good general approach might be to use the most general type that has the capabilities exercised by the implementation. In this case, we use `__getitem__` and `__len__`, which suggests [`Sequence`](https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes) might be appropriate. `numpy.ndarray` does not implement one of the required methods of Sequence (`.index()`), but we don't use that anyway. So, an appropriate type here might be something like `Union[Sequence[float], NDArray]`.

Reply:
Changed this in ff5c69b